sdk/src/test/scala/com/horizen/block/MainchainBlockReferenceTest.scala

```diff
@@ -963,6 +988,136 @@ class MainchainBlockReferenceTest extends JUnitSuite {
+
+    val scIdHex = "1aabf493f8605e78177d71c0aef3b251b2b9aaa66715fc86729a4e86ce566d8e"
+    val scId = new ByteArrayWrapper(BytesUtils.reverseBytes(BytesUtils.fromHexString(scIdHex))) // LE
+    val params = RegTestParams(scId.data)
+    val mcblockTry = MainchainBlockReference.create(mcBlockBytes, params, TestSidechainsVersionsManager(params))
+
+    assertTrue("Block expected to be parsed", mcblockTry.isSuccess)
+    val mcblock = mcblockTry.get
+
+    assertTrue("Block expected to be semantically valid", mcblock.semanticValidity(params).isSuccess)
+
+    val bytes = MainchainBlockReferenceDataSerializer.toBytes(mcblock.data)
+    var bytesFromFile:Array[Byte] = null
+    try {
+      val classLoader = getClass.getClassLoader
+      val file = new FileReader(classLoader.getResource("new_mc_blocks/mc_block_reference_2_certificates").getFile)
+      bytesFromFile = BytesUtils.fromHexString(new BufferedReader(file).readLine())
+    }
+    catch {
+      case e: Exception =>
+        jFail(e.toString)
+    }
+
+    assertArrayEquals("Serialized and regression bytes expected to be equal", bytesFromFile, bytes)
+
+    val deserializedMcBlockReferenceDataTry = MainchainBlockReferenceDataSerializer.parseBytesTry(bytes)
+    assertTrue("MainchainBlockReferenceData expected to by parsed.", deserializedMcBlockReferenceDataTry.isSuccess)
+    val mcBlockReferenceData = deserializedMcBlockReferenceDataTry.get
+
+    assertEquals("Block reference data must contain 1 top quality certificate.",
+      mcblock.data.topQualityCertificates.size, mcBlockReferenceData.topQualityCertificates.size)
+    assertEquals("Block reference data must contain 1 low quality certificate.",
+      mcblock.data.lowerCertificateLeaves.size, mcBlockReferenceData.lowerCertificateLeaves.size)
+    assertTrue("Block reference data must contain proof of existence.", mcBlockReferenceData.existenceProof.isDefined)
+    assertTrue("Block reference data must not contain proof of absence.", mcBlockReferenceData.absenceProof.isEmpty)
+  }
+
+  @Test
+  def mcBlockReferenceDataSerializationEmptySidechains(): Unit = {
+
+    val mcBlockHex = Source.fromResource("new_mc_blocks/mc_block_empty_sidechains").getLines().next()
+    val mcBlockBytes = BytesUtils.fromHexString(mcBlockHex)
+
+
+    // Test: Check for existing sidechain
+    val scIdHex = "0000000000000000000000000000000000000000000000000000000000000000" // valid FieldElement in BigEndian
+    val scId = new ByteArrayWrapper(BytesUtils.fromHexString(scIdHex))
+
+    val params = RegTestParams(scId.data)
+
+    val mcblockTry1 = MainchainBlockReference.create(mcBlockBytes, params, TestSidechainsVersionsManager(params))
+
+    assertTrue("Block expected to be parsed", mcblockTry1.isSuccess)
+    val mcblock1 = mcblockTry1.get
+
+    assertTrue("Block expected to be semantically valid", mcblock1.semanticValidity(params).isSuccess)
+
+    val bytes = MainchainBlockReferenceDataSerializer.toBytes(mcblock1.data)
```

Comment:
no need to parse original mc block ref data.
Instead:
1) parse mc block ref data from regression file
2) check that it was successful
3) check the content of the data (as you already do below)
4) to a round trip check: serialize mc block ref data back to bytes and check to be equal to the bytes from regression file.

Same in 2 other regression tests